c++/triqs_Nevanlinna/CMakeLists.txt

```diff
@@ -31,11 +31,13 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DESTINATION include FILES_MATCHING
 
 # ========= Library Dependencies ==========
 
-find_package(Eigen3 REQUIRED)
+find_package(Eigen3 REQUIRED CONFIG)
 target_link_libraries(${PROJECT_NAME}_c PUBLIC Eigen3::Eigen)
+target_compile_definitions(${PROJECT_NAME}_c PUBLIC EIGEN_HAS_STD_RESULT_OF=0)
```

Comment:
What's the problem if you don't put this?

Reply:
Prior to Eigen 3.4, the template `Eigen::internal::result_of` attempts to use `std::result_of` which was removed in C++20 in favor of `std::invoke_result`. However, Eigen also has its own implementation of `result_of` which can be used by simply defining this macro. It doesn't interfere with newer versions of Eigen. These will still use `std::invoke_result` because guarded by yet another feature macro.
https://gitlab.com/libeigen/eigen/-/commit/a31effc3bca2f0924752caeebfd6f61f7edf9a43